src/vm/codeman.cpp

```diff
@@ -1304,67 +1305,109 @@ void EEJitManager::SetCpuInfo()
+        //    PCLMULQDQ - ECX bit 25   (buffer[11] & 0x01)
         // CORJIT_FLAG_USE_BMI1 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
         //    BMI1 - EBX bit 3     (buffer[4]  & 0x08)
         // CORJIT_FLAG_USE_BMI2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
         //    BMI2 - EBX bit 8     (buffer[5]  & 0x01)
         // CORJIT_FLAG_USE_LZCNT if the following feature bits are set (input EAX of 80000001H)
         //    LZCNT - ECX bit 5    (buffer[8]  & 0x20)
         // synchronously updating VM and JIT.
+
         (void) getcpuid(1, buffer);
-        // If SSE2 is not enabled, there is no point in checking the rest.
-        // SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
+
+        // If SSE/SSE2 is not enabled, there is no point in checking the rest.
+        //   SSE  is bit 25 of EDX   (buffer[15] & 0x02)
+        //   SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
         // TODO: Determine whether we should break out the various SSE options further.
-        if ((buffer[15] & 0x04) != 0)               // SSE2
+        if ((buffer[15] & 0x06) != 0x06)                                    // SSE & SSE2
         {
-            if ((buffer[8]  & 0x01) != 0)           // SSE3
+            if ((buffer[8]  & 0x01) != 0)                                   // SSE3
             {
                 CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE3);
-                if ((buffer[9]  & 0x02) != 0)           // SSSE3
+
+                if ((buffer[9]  & 0x02) != 0)                               // SSSE3
                 {
                     CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSSE3);
-                    if ((buffer[10] & 0x08) != 0)           // SSE4.1
+
+                    if ((buffer[10] & 0x08) != 0)                           // SSE4.1
                     {
                         CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE41);
-                    }
-                    if ((buffer[10] & 0x10) != 0)           // SSE4.2
-                    {
-                        CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE42);
-                        if ((buffer[10] & 0x80) != 0)       // POPCNT
+
+                        if ((buffer[10] & 0x10) != 0)                       // SSE4.2
                         {
-                            CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_POPCNT);
+                            CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE42);
+
+                            if ((buffer[10] & 0x80) != 0)                   // POPCNT
+                            {
+                                CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_POPCNT);
+                            }
+
+                            if ((buffer[11] & 0x18) == 0x18)                // AVX
+                            {
+                                if(DoesOSSupportAVX() && (xmmYmmStateSupport() == 1))
```

Comment:
`DoesOSSupportAVX` just returns `true` for Linux/Mac, is this expected?

`xmmYmmStateSupport` does not do the required check:
> Detect CPUID.1:ECX.OSXSAVE[bit 27] = 1 (XGETBV enabled for application use1)

We should ensure that this check is done on Linux/Mac and validate whether or not `GetEnabledXStateFeatures` already does this check on Windows.